sample-app/src/main/java/com/joinforage/android/example/ui/pos/POSViewModel.kt

```diff
@@ -1,5 +1,48 @@
 package com.joinforage.android.example.ui.pos
 
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.joinforage.android.example.ui.pos.data.Merchant
+import com.joinforage.android.example.ui.pos.data.POSUIState
+import com.joinforage.android.example.ui.pos.network.PosApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import retrofit2.HttpException
 
-class POSViewModel : ViewModel()
+sealed interface MerchantDetailsState {
+    object Idle : MerchantDetailsState
+    data class Success(val merchant: Merchant) : MerchantDetailsState
+    data class Error(val error: String) : MerchantDetailsState
+    object Loading : MerchantDetailsState
```

Comment:
Not blocking at all - just curious. Why `data class`es for `Success` and `Error` and `Object`s for `Idle` and `Loading`?

Reply:
The Android tutorial/guide I went through suggested doing it this way. I think `data class` over `object` is so you can pass values into those, since `Success` and `Error` have data associated with them but `Idle` and `Loading` don't need that.